Meter.c

```diff
@@ -71,6 +72,12 @@ static void TextMeterMode_draw(Meter* this, int x, int y, int w) {
 static const char BarMeterMode_characters[] = "|#*@$%&.";
 
 static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
+   // Prevent (x + w) overflow
+   assert(x >= 0);
+   x = MAXIMUM(0, x);
+   assert(w <= INT_MAX - x);
+   w = MINIMUM(INT_MAX - x, w);
```

Comment:
I was considering whether this clamping of `x` and `w` is really necessary. While the two `assert` lines are useful, it might be easier to guard against overflow on the callers of the `Meter.draw()` functions, rather than guard on the callees.

There are only four functions that would compute `x`, `y`, and `w` values for each meter and call `Meter.draw()`:
* `Header_draw`
* `CPUMeterCommonDraw`
* `SingleColCPUsMeter_draw`
* `MemorySwapMeter_draw`